flink-core/src/test/java/org/apache/flink/api/common/state/StateDescriptorTest.java

```diff
@@ -224,6 +253,15 @@ public void testEqualsSameNameAndTypeDifferentClass() throws Exception {
 		public Type getType() {
 			return Type.VALUE;
 		}
+
+		public TypeSerializer<T> getOriginalSerializer() {
```

Comment:
This would not be necessary if we move this to the `StateDescriptor` class itself, and make the atomic reference field private. IMO, this is safer (please see my other comment above).